public/js/test.js

```diff
@@ -45,6 +45,22 @@ function doOauth() {
   window.open("/oauth/init");
 }
 
+function isValidEmail(val) {
+  // https://stackoverflow.com/a/46181
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(String(val).toLowerCase());
+}
+
+function enableBtnIfEmailValid(e) {
+  const emailBtn = document.querySelector("#subscribe-email-btn");
```

Comment:
getElementById is three orders of magnitude faster than querySelector: https://jsperf.com/nhnt11-queryselector-vs-getelementbyid

Since this is being called continuously with input, we should use getElementById. I'd even say we could just cache emailBtn outside the function to avoid accessing the DOM every time.

Reply:
updated to `getElementById`